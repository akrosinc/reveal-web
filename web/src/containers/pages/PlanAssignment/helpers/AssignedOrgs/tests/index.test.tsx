import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import {
  organization1,
  organization2,
  organization3,
} from '../../../../../../store/ducks/tests/fixtures';
import { AssignedOrgs } from '../index';

describe('PlanAssignment/AssignedOrgs', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders when no orgs', () => {
    const wrapper = mount(<AssignedOrgs displayLimit={2} id="42" orgs={[]} />);
    expect(toJson(wrapper.find('span'))).toMatchSnapshot('Org names');
    expect(wrapper.find('Tooltip').length).toEqual(0);
    wrapper.unmount();
  });

  it('renders when limit greater than orgs', () => {
    const wrapper = mount(
      <AssignedOrgs displayLimit={6} id="42" orgs={[organization1, organization2, organization3]} />
    );
    expect(toJson(wrapper.find('span'))).toMatchSnapshot('Org names');
    expect(wrapper.find('Tooltip').length).toEqual(0);
    wrapper.unmount();
  });

  it('renders when limit lower than orgs', async () => {
    const div = document.createElement('div');
    div.setAttribute('id', 'org-tooltip-42');
    document.body.appendChild(div);

    const wrapper = mount(
      <AssignedOrgs displayLimit={2} id="42" orgs={[organization1, organization2, organization3]} />
    );
    expect(toJson(wrapper.find('span'))).toMatchSnapshot('Org names');
    expect(wrapper.find('Tooltip').length).toEqual(1);

    wrapper.unmount();
  });
});
