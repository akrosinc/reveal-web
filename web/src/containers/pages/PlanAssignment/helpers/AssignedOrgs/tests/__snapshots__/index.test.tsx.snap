// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanAssignment/AssignedOrgs renders when limit greater than orgs: Org names 1`] = `
<span
  id="org-span-42"
>
  The Luang, Demo Team, Takang 1
</span>
`;

exports[`PlanAssignment/AssignedOrgs renders when limit lower than orgs: Org names 1`] = `
<span
  id="org-tooltip-42"
>
  The Luang, Demo Team and 1 more
</span>
`;

exports[`PlanAssignment/AssignedOrgs renders when no orgs: Org names 1`] = `null`;
